### ISSUE TITLE
Felguard / Pet Simulation Accuracy issues

### ISSUE THREAD

**Issue by scamille** [scamille/simc_issue_test#22]
_From [stephen....@gmail.com](https://code.google.com/u/111476282210466734811/) on January 26, 2009 18:04:37_

What steps will reproduce the problem? 1. Run Simulationcraft and pull the Count numbers for the pet in question
2. Combatlog a fight and sum the Hit/Crit/Miss/Dodge/Parry/Glance #'s and 
divide that number by the time of the fight. What is the expected output? What do you see instead? I'm expecting the time to be near 6 seconds for cleave and much lower for 
hit attacks for a felguard-

I'll work with the following WWS and go through the oddities between 
simulation and results to hopefully help you pin down the problem.

From this WWS- http://wowwebstats.com/ofqodetjtxz6m?a=x252deab&s=101006-112770 Time Duration = 160~ seconds

Cleave attempts = 28
Hit attempts = 125

The example simulation puts cleave at 6.2 seconds whereas this WWS has 
cleave occuring on averazge 160/28 5.71 (6 second cooldown).  Which is 
consistent with the last cleave being cast near or as the killing blow and 
not spending any time on cooldown for the last cleave.

The example simulation puts melee attacks occuring every 1.5 seconds, but 
the resulting WWS has melee attacks occuring every 160/125 = 1.28 seconds.

The Felguard may be off by a good 20&#37; for melee swings and 1-2&#37; for 
cleaves.  This may affect other pets white damage as well.  These 
discrepencies seem consistent with other WWS logs for other fights.

*note - There was no heroism on this fight. What version of the product are you using? On what operating system? Please provide any additional information below.

_Original issue: http://code.google.com/p/simulationcraft/issues/detail?id=22_

**Comment by scamille** [comment]
_From [jon.valv...@gmail.com](https://code.google.com/u/104910199056440286473/) on January 26, 2009 20:53:52_

I believe you're misinterpreting the WWS output. Unless I'm mistaken, glancing blows 
are counted among the hits, and the "all miss" column is just the sum of 
miss/parry/dodge/etc. With those assumptions, I see 98 melee swings, which a bit over 
1.6 swings per second. Which sounds reasonable given no heroism.

I believe the cleave rate discrepancy you're seeing may be due to the lag parameter - 
I'll investigate.

**Status:**

**Comment by scamille** [comment]
_From [jon.valv...@gmail.com](https://code.google.com/u/104910199056440286473/) on January 26, 2009 21:34:40_

And when I say swings per second, I do of course mean seconds per swing.

**Comment by scamille** [comment]
_From [jon.valv...@gmail.com](https://code.google.com/u/104910199056440286473/) on January 26, 2009 21:54:44_

Ok, from what I can see - and I'm sure Nate will step in and correct me if I'm wrong 
- pets are treated like regular players when it comes to lag. Judging from your WWS, 
this is incorrect behavior when it comes to the Felguard's cleave, and probably other 
pet abilities with a cooldown as well, like the Felhunter's shadow bite.

There is an (undocumented?) option called pet_lag which you can use to override the 
lag parameter for just pets. This isn't a good solution to the problem, though, 
because there are pet abilities that definitely do have lag, such as the Imp's fire 
bolt.

So in summary: Yes, we probably need to look at how lag is implemented for various 
pet abilities. It may be as simple as removing lag for abilities with a cooldown.

Either way it doesn't look like this issue has a huge impact on accuracy.

**Status:** Accepted

**Comment by scamille** [comment]
_From [natehieter@gmail.com](https://code.google.com/u/natehieter@gmail.com/) on January 27, 2009 05:17:49_

I'll create a "lag" field in player_t that inherits from the appropriate sim_t field,
which will allow us to make lag player/pet specific.

**Comment by scamille** [comment]
_From [natehieter@gmail.com](https://code.google.com/u/natehieter@gmail.com/) on February 09, 2009 10:45:02_

Considerable work was done to improve Warlock Pet accuracy.
Moving state to "fixed".

**Status:** Fixed
